Source/Core/Core/PrimeHack/Mods/FpsControls.cpp

```diff
@@ -127,7 +147,151 @@ void FpsControls::run_mod_menu(Region region) {
   }
 }
 
+bool FpsControls::beam_visor_menu_handler(u32 cursor_base, u32 yaw_vel_address, Game game)
+{
+  if (CheckBeamMenuCtl() || CheckVisorMenuCtl())
+  {
+    if (menu_cursor_state == Beam_Visor_Menu_State::IDLE)
+    {
+      set_cursor_pos(0, 0);
+    }
+    handle_cursor(cursor_base + 0x9c, cursor_base + 0x15c, 0.95f, 0.90f);
+    write32(0, yaw_vel_address);
+
+    // Set the beam/visor menu state.
+    menu_cursor_state = CheckBeamMenuCtl() ? Beam_Visor_Menu_State::SET_BEAM : Beam_Visor_Menu_State::SET_VISOR;
+    return true;
+  }
+
+  else if (menu_cursor_state == Beam_Visor_Menu_State::SET_BEAM || menu_cursor_state == Beam_Visor_Menu_State::SET_VISOR)
+  {
+    determine_selected_beam_visor(game, menu_cursor_state);
+
+    set_cursor_pos(0, 0);
+    write32(0, cursor_base + 0x9c);
+    write32(0, cursor_base + 0x15c);
+    menu_cursor_state = Beam_Visor_Menu_State::IDLE;
+  }
+
+  return false;
+}
+
+
+/*
+Since Prime 1 & 2 have part of their usual beam selection code overridden
+(in order to make the beam selection buttons work), the in-game beam selection menu
+no longer actually equips the selected beam. I've reimplemented that here.
+
+The numbers used in the equations below were the values (in memory) for the cursor position
+at the boundries between panels, using a 16:9 aspect ratio.
+*/
+void FpsControls::determine_selected_beam_visor(Game game, Beam_Visor_Menu_State beam_visor)
+{
+  if (beam_visor == Beam_Visor_Menu_State::IDLE) {
+    return;
+  }
+
+  bool beam = beam_visor == Beam_Visor_Menu_State::SET_BEAM;
+
+  float x = get_cursor_x();
+  float y = get_cursor_y();
+
+
+  //The center is the area inside an oval: 1 > (x/major-axis)^2 + (y/minor-axis)^2
+  if (1 > pow((x / 0.136), 2) + pow((y / 0.2), 2)) {
+    // Select beam 1;
+    if (beam) {
+      request_beam_change(0);
+    }
+    else {
+      request_visor_change(0);
+    }
+    
+
+  }
+
+  // The upper section is a V shape (minus the center oval): y < m * abs(x) + b
+  else if (y < -(0.78667 / 0.95) * abs(x))
```

Comment:
Another example of too much empty lines. 

Try to stay consistent with your formatting, some lines else-if statements have the bracket on the same line, others do not.